pkg/piecestore/rpc/server/psdb/psdb.go

```diff
@@ -247,3 +253,54 @@ func (db *DB) DeleteTTLByID(id string) error {
 	}
 	return err
 }
+
+// AddBandwidthUsed adds bandwidth usage into database by date
+func (db *DB) AddBandwidthUsed(size int64, t time.Time) (err error) {
+	defer db.locked()()
+
+	unixtimenow := t.Unix()
+	daystartunixtime := time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, t.Location()).Unix()
+	dayendunixtime := time.Date(t.Year(), t.Month(), t.Day(), 24, 0, 0, 0, t.Location()).Unix()
+
+	var getSize int64
+	if (unixtimenow >= daystartunixtime) && (unixtimenow <= dayendunixtime) {
+		err = db.DB.QueryRow(`SELECT size FROM bwusagetbl WHERE daystartdate <= ? AND ? <= dayenddate`, unixtimenow, unixtimenow).Scan(&getSize)
+		switch {
+		case err == sql.ErrNoRows:
+			_, err = db.DB.Exec("INSERT INTO bwusagetbl (size, daystartdate, dayenddate) VALUES (?, ?, ?)", size, daystartunixtime, dayendunixtime)
+			return err
+		case err != nil:
+			return err
+		default:
+			getSize = size + getSize
+			_, err = db.DB.Exec("UPDATE bwusagetbl SET size = ? WHERE daystartdate = ?", getSize, daystartunixtime)
+			return err
+		}
+	}
+	return err
+}
+
+// GetBandwidthUsedByDay finds the so far bw used by day and return it
+func (db *DB) GetBandwidthUsedByDay(t time.Time) (size int64, err error) {
+	defer db.locked()()
+
+	daystarttime := time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, t.Location()).Unix()
+	err = db.DB.QueryRow(`SELECT size FROM bwusagetbl WHERE daystartdate=?`, daystarttime).Scan(&size)
+	return size, err
+}
+
+// GetTotalBandwidthBetween each row in the bwusagetbl contains the total bw used per day
+func (db *DB) GetTotalBandwidthBetween(startdate time.Time, enddate time.Time) (totalbwusage int64, err error) {
+	defer db.locked()()
+
+	startTimeUnix := time.Date(startdate.Year(), startdate.Month(), startdate.Day(), 0, 0, 0, 0, startdate.Location()).Unix()
+	endTimeUnix := time.Date(enddate.Year(), enddate.Month(), enddate.Day(), 0, 0, 0, 0, enddate.Location()).Unix()
+	defaultunixtime := time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day(), 0, 0, 0, 0, time.Now().Location()).Unix()
+
+	if (endTimeUnix < startTimeUnix) && (startTimeUnix > defaultunixtime || endTimeUnix > defaultunixtime) {
+		return totalbwusage, errors.New("Invalid date range")
+	}
+
+	err = db.DB.QueryRow(`SELECT SUM(size) FROM bwusagetbl WHERE daystartdate BETWEEN ? AND ?`, startTimeUnix, endTimeUnix).Scan(&totalbwusage)
```

Comment:
Just want to double check that the SUM doesn't have any issues when the table is empty

Reply:
if so we can just check if the table is empty before running the sum

Reply:
Tested the sum with no columns and it returns zero. Also QueryRow handles no rows.